### PR TITLE
Add support for WakuMessage and merge publish function

### PR DESCRIPTION
**Pull request by oskarth**
Fixes https://github.com/status-im/nim-waku/issues/118

Partially addresses https://github.com/status-im/nim-waku/issues/121

Builds on https://github.com/status-im/nim-waku/pull/123

---

Running examples we can see:

````
DBG 2020-09-01 13:32:43+08:00 publish                                    tid=24041 topic=WakuRelay contentTopic=foo
INF 2020-09-01 13:32:43+08:00 Hit subscribe handler                      tid=24041 topic=foobar payload="hello world" contentTopic=foo
```

### CODE REVIEW

File: waku/node/v2/waku_types.nim [hunk 1]
```diff
@@ -16,7 +16,30 @@ type
...
-    messages*: seq[(Topic, Message)]
+  # TODO Revist messages field indexing as well as if this should be Message or WakuMessage
+    messages*: seq[(Topic, WakuMessage)]
+
+  WakuMessage* = object
+    payload*: seq[byte]
+    contentTopic*: string
+
+
+
```
Comment (decanus) [inline waku/node/v2/waku_types.nim:29]: Seems like an excess of new lines

**Review by decanus — CHANGES_REQUESTED** [review]

File: waku/node/v2/waku_types.nim (hunk 1 above)
Comment (oskarth) [reply waku/node/v2/waku_types.nim:29]: Deliberate to space out different parts of the code, like sections. Makes it easier to visually scan IMO. Also makes code folding look better.

Comment (oskarth) [inline waku/node/v2/waku_types.nim:29]:
```suggestion
    contentTopic*: string

```

Comment (oskarth) [reply waku/node/v2/waku_types.nim:29]: If it helps we can remove one or two

**Comment by oskarth** [comment]
@decanus do you have any comments on the content of the PR?

**Comment by decanus** [comment]
@oskarth other than the comment left, lgtm.

**Review by decanus — APPROVED** [review]

File: docs/api/v2/node.md [hunk 2]
```diff
@@ -46,19 +46,12 @@ method unsubscribe*(w: WakuNode, contentFilter: ContentFilter)
   ## Status: Not yet implemented.
   ## TODO Implement.
 
-method publish*(w: WakuNode, topic: Topic, message: Message)
-  ## Publish a `Message` to a PubSub topic.
+method publish*(w: WakuNode, topic: Topic, message: WakuMessage)
```
Comment (kdeme) [inline docs/api/v2/node.md:49]: Looks fine for now but eventually we should probably get something where topic is optional and some default (set at init?) is used. Or something better...

File: waku/node/v2/wakunode2.nim [hunk 3]
```diff
@@ -176,30 +177,24 @@ proc unsubscribe*(w: WakuNode, contentFilter: ContentFilter) =
...
-  ##
-  ## Status: Not yet implemented.
-  ## TODO Implement as wrapper around `waku_relay` and `publish`.
-  ## TODO WakuMessage. Ensure content filter is in it.
+  # TODO Basic getter function for relay
+  let wakuRelay = cast[WakuRelay](node.switch.pubSub.get())
 
-  w.messages.insert((contentFilter.contentTopic, message))
+  # XXX Unclear what the purpose of this is
+  # Commenting out as it is later expected to be Message type, not WakuMessage
```
Comment (kdeme) [inline waku/node/v2/wakunode2.nim:193]: Yeah, I guess this is only there for historical messages? So this should eventually be replaced with an optional (when historical messages is enabled) call to store the messages.

**Review by kdeme — APPROVED** [review]
lgtm

**Comment by kdeme** [comment]
I went ahead on rebasing & fixing test compile issue here (+ disable outdated test) as else I'm continuing with more and more merge conflicts in the part I'm working on.

File: docs/api/v2/node.md (hunk 2 above)
Comment (oskarth) [reply docs/api/v2/node.md:49]: Wouldn't this be a convenience function defined in e.g. status-waku? E.g. a partial function defined by config default topic.

File: waku/node/v2/wakunode2.nim (hunk 3 above)
Comment (oskarth) [reply waku/node/v2/wakunode2.nim:193]: Do we want Message or WakuMessage as part of historical messages? Not obvious to me.

cc @decanus

**Comment by oskarth** [comment]
Thanks @kdeme, forgot to adjust those tests!

**Pull request merged**